drivers/include/at86rf2xx.h

```diff
@@ -204,36 +112,129 @@ extern "C" {
 #define AT86RF2XX_OPT_AUTOACK        (0x0080)       /**< Auto ACK active */
 #define AT86RF2XX_OPT_ACK_PENDING    (0x0100)       /**< ACK frames with data
                                                      *   pending */
-
 /** @} */
 
-#if defined(MODULE_AT86RFA1) || defined(MODULE_AT86RFR2)
-/**
- * @brief   memory mapped radio needs no parameters
- */
-typedef void at86rf2xx_params_t;
-#else
 /**
  * @brief   struct holding all params needed for device initialization
  */
 typedef struct at86rf2xx_params {
+#if AT86RF2XX_NEED_SPI || defined(DOXYGEN)
     spi_t spi;              /**< SPI bus the device is connected to */
     spi_clk_t spi_clk;      /**< SPI clock speed to use */
     spi_cs_t cs_pin;        /**< GPIO pin connected to chip select */
+#endif
     gpio_t int_pin;         /**< GPIO pin connected to the interrupt pin */
     gpio_t sleep_pin;       /**< GPIO pin connected to the sleep pin */
     gpio_t reset_pin;       /**< GPIO pin connected to the reset pin */
 } at86rf2xx_params_t;
-#endif
 
 /**
  * @brief   Device descriptor for AT86RF2XX radio devices
  *
  * @extends netdev_ieee802154_t
  */
-typedef struct {
+typedef struct at86rf2xx_base {
     netdev_ieee802154_t netdev;             /**< netdev parent struct */
-#if defined(MODULE_AT86RFA1) || defined(MODULE_AT86RFR2)
+    uint16_t flags;                         /**< Device specific flags */
+    uint8_t state;                          /**< current state of the radio */
+    uint8_t tx_frame_len;                   /**< length of the current TX frame */
+    uint8_t idle_state;                     /**< state to return to after sending */
+    uint8_t pending_tx;                     /**< keep track of pending TX calls
+                                                 this is required to know when to
+                                                 return to @ref at86rf2xx_t::idle_state */
+    uint8_t dev_type;                      /**<  AT86RF2XX device type */
+} at86rf2xx_base_t;
+
+/**
+ * @brief An pointer to an instance of an actual AT86RF2XX
+ * must safely be casted to a pointer of this type
+ */
+typedef struct at86rf2xx {
+    at86rf2xx_base_t base;
+    at86rf2xx_params_t params;
+} at86rf2xx_t;
+
+#if IS_USED(MODULE_AT86RF212B)
+typedef struct at86rf212b_params {
```

Comment:
I think variant specific descriptors should be in their respective header file (e.g this one in the at86rf212b.h file)

Reply:
Schould e.g. `at86212b_t` and `at86rf212b_params_t` and `at86rf212b_setup()` be publicly available or just within the driver, e.g in `at86rf2xx.c`?
I decided to have them publicly available because if someone knows that he builds an application just for `at86212b` he could directly call `at86rf212b_setup()`